mux/doc.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2012 The Gorilla Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+Package mux implements a request router and dispatcher.
```

Comment:
Can we move this into `internal/mux/` just so no outside dependencies can happen? Or is it going to be a problem with the runtime exposing it's mux? 🤔 I'm not sure I remember the details right.

Reply:
While I have not considered adding it to internal, I have had the same question regarding the plugins package. While the router member is private on Manager `Manager.r`, the package does expose the `WithRouter(r *mux.Router) func(*Manager) {...}` function.
Might also apply to Server with its `WithRouter` method as well(?).

Not exactly sure about the usage of server and runtime outside of OPA itself though.

Not sure about the usage of Server and